src/downloadmanager.cpp

```diff
@@ -916,7 +937,7 @@ void DownloadManager::resumeDownloadInt(int index)
 
   // Check for finished download;
   if (info->m_TotalSize <= info->m_Output.size() && info->m_Reply != nullptr
-      && info->m_Reply->isOpen() && info->m_Reply->isFinished() && info->m_State != STATE_ERROR) {
+      && info->m_Reply->isFinished() && info->m_State != STATE_ERROR) {
```

Comment:
Any reason for this change?

Reply:
Correct, this is actually a leftover from investigating the cache issue, I used `resumeDownloadInt`
 as a crutch. When MO2 is closed, file exists, and cache enabled, `resumeDownloadInt` would skip this block, because the connection is closed at this point, and append the same contents to the file again, doubling its size.
Since `resumeDownloadInt` is irrelevant to this PR, I have reverted this change.

Reply:
Correction - this is unrelated to the cache issue, happens with disabled cache as well, I brought the change back.